rllib/models/tf/tf_action_dist.py

```diff
@@ -88,7 +88,7 @@ def kl(self, other: ActionDistribution) -> TensorType:
 
     @override(TFActionDistribution)
     def _build_sample_op(self) -> TensorType:
-        return tf.squeeze(tf.random.categorical(self.inputs, 1), axis=1)
+        return tf.squeeze(tf.random.categorical(tf.math.log(tf.nn.softmax(tf.where(self.inputs!=0, self.inputs, tf.float64.min))), 1), axis=1)
```

Comment:
please use a local varialbe:
```
logp = tf.math.log(tf.nn.softmax(tf.where(self.inputs!=0, self.inputs, tf.float64.min)))
tf.squeeze(tf.random.categorical(logp, 1), axis=1)
```

Reply:
I also don't get why we need to modify the logits here, shouldn't we fix action_masking to use tf.float64.min instead?

Reply:
there's 2 reasons to modify the logits here:
1. leaving a lot of (what is essentially) -inf around can cause nan problems with algos like ppo
2. action_masking doesn't catch all cases (e.g. people will do it some other way or have a more complex situation)